src/feature.js

```diff
@@ -122,12 +122,31 @@ class Feature {
 
       data.completeLocation = null;
       tabInfo.currentOrigin = data.origin;
-      if (tabInfo.currentOrigin !== tabInfo.currentOriginReported) {
+
+      if (tabInfo.currentOriginReported &&
+          tabInfo.currentOrigin !== tabInfo.currentOriginReported &&
+          !tabInfo.waitingForReturn) {
+        // If the user does not return within 15 min, clear the data, the user has ignored the banner.
+        tabInfo.bannerTimer = setTimeout(function() {
+          tabInfo.payloadWaitingForSurvey = null;
+          tabInfo.currentOriginReported = null;
+          tabInfo.waitingForReturn = null;
```

Comment:
What would happen if we navigated to a.com, triggered the notification, then navigated to b.com, and triggered the notification again? Would the first timeout set the waitingForReturn to null, cancelling out the second timeout? (Or put another way, do we need the waitingForReturn to keep a list of origins it's waiting for, and when to check again?)

Reply:
Thank you, That was something we decided on, and I should have made note of in the PR description. We want a.com's timer to be cancelled, and b.com to overwrite it. I don't want to keep track of multiple waiting reports at once. I don't think I remembered to cancel a.com's timer though, will add that in.